# This workflow calls the reusable test workflow and ensures that the
# tests do not pass.
#
# Please don't use this as a reusable workflow, because it is very
# much not.

name: Internal job invocation that expects a test failure result
on:
  workflow_call:
    inputs:
      module_base:
        description: "Directory containing the go.mod of the codebase under test"
        type: string
        default: "."
      go_version:
        description: "Go toolchain to use"
        type: string

jobs:
  tests:
    uses: "./.github/workflows/ci_baseline_go_tests.yml"
    with:
      module_base: ${{inputs.module_base}}
      go_version: ${{inputs.go_version}}
      _internal_continue_on_error: true

  expect_test_failure:
    runs-on: ubuntu-latest
    needs: tests
    steps:
      - name: transform expected failure
        id: expected_failure
        env:
          NEEDS_JSON: ${{toJSON(needs)}}
          JQ_FAIL_PROGRAM: >
            .tests.outputs["_internal_tests_result"] == "failure"
        run: >
          echo "did_fail=$(echo "$NEEDS_JSON" | jq -r "$JQ_FAIL_PROGRAM")" | tee -a $GITHUB_OUTPUT
      - name: expect failure
        run: exit 1
        if: steps.expected_failure.outputs.did_fail != 'true'
